composer/trainer/trainer.py

```diff
@@ -2733,7 +2737,10 @@ def _train_microbatches(
                         optimizer.zero_grad()
 
             # Tracker for gradient accumulation
-            current_batch_size = sum([self._train_data_spec.get_num_samples_in_batch(batch) for batch in microbatches])
+            if self.accumulate_train_batch_on_tokens:
+                current_batch_size = sum([self._train_data_spec.get_num_tokens_in_batch(b) for b in microbatches])
```

Comment:
Can we throw a warning if this is 0? Aka you use a default data spec which fails to parse number of tokens